src/http_proxy_io.c

```diff
@@ -1025,18 +1025,29 @@ static OPTIONHANDLER_HANDLE http_proxy_io_retrieve_options(CONCRETE_IO_HANDLE ht
             OPTIONHANDLER_HANDLE underlying_io_options;
 
             /* Codes_SRS_HTTP_PROXY_IO_01_046: [ http_proxy_io_retrieve_options shall return an OPTIONHANDLER_HANDLE obtained by calling xio_retrieveoptions on the underlying IO created in http_proxy_io_create. ]*/
-            if ((underlying_io_options = xio_retrieveoptions(http_proxy_io_instance->underlying_io)) == NULL ||
-                OptionHandler_AddOption(result, OPTION_UNDERLYING_IO_OPTIONS, underlying_io_options) != OPTIONHANDLER_OK)
+            if ((underlying_io_options = xio_retrieveoptions(http_proxy_io_instance->underlying_io)) == NULL)
             {
                 /* Codes_SRS_HTTP_PROXY_IO_01_048: [ If xio_retrieveoptions fails, http_proxy_io_retrieve_options shall return NULL. ]*/
-                LogError("unable to save underlying_io options");
-                OptionHandler_Destroy(underlying_io_options);
+                LogError("unable to retrieve underlying_io options");  
                 OptionHandler_Destroy(result);
                 result = NULL;
             }
-            else
+            else 
             {
-                // All is fine.
+                if (OptionHandler_AddOption(result, OPTION_UNDERLYING_IO_OPTIONS, underlying_io_options) != OPTIONHANDLER_OK)
+                {
+                    /* Codes_SRS_HTTP_PROXY_IO_01_048: [ If xio_retrieveoptions fails, http_proxy_io_retrieve_options shall return NULL. ]*/
+                    LogError("unable to save underlying_io options");
+                    OptionHandler_Destroy(result);
+                    result = NULL;
+                }
+                else
```

Comment:
I think this should be removed as it doesn't do anything and it's confusing. (Comment is OK to stay.)

Reply:
I hear you.  However, it's how it's done in (I think) every other file that has this same section of logic.  I can go either way.

Reply:
Thanks! I see it now. 
It does seem to be part of the coding standards for the repo so it's OK to follow the same pattern.